core/shared/src/main/scala/org/http4s/internal/Logger.scala

```diff
@@ -46,12 +47,18 @@ object Logger {
       val isJson = message.contentType.exists(mT =>
         mT.mediaType == MediaType.application.json || mT.mediaType.subType.endsWith("+json")
       )
-      val bodyStream = if (!isBinary || isJson) {
-        message.bodyText(implicitly, message.charset.getOrElse(Charset.`UTF-8`))
-      } else {
-        message.body.map(b => java.lang.Integer.toHexString(b & 0xff))
-      }
-      Some(bodyStream.compile.string)
+      val string =
+        if (!isBinary || isJson)
+          message
+            .bodyText(implicitly, message.charset.getOrElse(Charset.`UTF-8`))
+            .compile
+            .string
+        else
+          message.body.compile
+            .to(fs2.Collector.supportsByteVector(ByteVector))
+            .map(_.toHex)
```

Comment:
Very nice! 👏

Reply:
Actually this seems a big verbose. `supportsByteVector` is an implicit, there should be another way to use it? 🤔

Reply:
Don't you mind if I put your handle in the co-authored commit? Just a golden finding. I'm thrilled about how we reduce memory consumption for this one.

Reply:
It'd be nice if `fs2` could expose the `byteVector` method like they have `fs2.Collector#string`. Otherwise, I have no other ideas 🤷🏻

Reply:
By the way, we use `supportsByteVector` this way in ember already https://github.com/http4s/http4s/blob/series/0.23/ember-core/shared/src/main/scala/org/http4s/ember/core/h2/H2Server.scala#L188.

Reply:
> It'd be nice if `fs2` could expose the `byteVector` method like they have `fs2.Collector#string`. Otherwise, I have no other ideas 🤷🏻

Oh is it an omission? We should fix that

Reply:
It'd be a good addition to fs2 :+1:

Reply:
Does this one not work?
https://github.com/typelevel/fs2/blob/7af8d5d02499b2a74751af913947afd0282a2b40/core/shared/src/main/scala/fs2/Collector.scala#L101

Reply:
Hmm, it's a `Builder`, and not `Collector`. `fs2.Stream.CompileOps#to` requires `Collector` in arguments.

Reply:
Oh sorry, you are right! Lets' get that fixed, thanks :)

Reply:
I'm really confused, it's even documented here
https://github.com/typelevel/fs2/blob/7af8d5d02499b2a74751af913947afd0282a2b40/core/shared/src/main/scala/fs2/Stream.scala#L4920

Reply:
Instead of trying to write `stream.compile.to[String]`, you can just write `.compile.string`, using this method. https://github.com/typelevel/fs2/blob/main/core/shared/src/main/scala/fs2/Stream.scala#L4895-L4896